### ISSUE TITLE
Refactor for the deprecation of FTP?

### ISSUE THREAD

**Issue by noah-de** [nasa/podaacpy#136]
Should we make issues for each area to be refactored with the imminent deprecation of FTP?
https://podaac.jpl.nasa.gov/announcements/2019-05-06_REMINDER_PO.DAAC_FTP_RETIREMENT

**Comment by lewismc** [comment]
Hi @noah-de yes we most likely should. There is one way for us to do this... essentially we could wait until FTP is totally dropped at PO.DAAC, run our unit tests and fix any broken code such that tests pass. wdyt?

**Comment by noah-de** [comment]
Very true @lewismc a common approach is to _wait and see what breaks_.  However, I thought that you might have some knowledge from either NASA or JPL. Either from some high level documentation on the podaacpy's dependencies of PO.DAAC or conversations around the water cooler.
I would prefer to be proactive: retire FTP code and leverage PO.DAAC Drive.
As long as it is fixed before the next release, it probably makes no difference.

**Comment by lewismc** [comment]
I'll wait another week or so until FTP has been totally deprecated before releasing again @noah-de

**Comment by lewismc** [comment]
@noah-de any comments here?

**Comment by noah-de** [comment]
Not at the moment. I am looking forward to seeing what we can learn from the new health check introduced by @sanketsaurav in #139.

The build still fails (https://travis-ci.org/nasa/podaacpy/builds/566054604) and I have not had time to dive into the cause and resolve it.

I leave it to your discretion.

**Comment by lewismc** [comment]
I *think* that MCC functionality is currently broken. I'll work on that.
I'm spending some time today and will submit pull requests as I go. Thanks

**Comment by noah-de** [comment]
Terrific!
Thanks

**Comment by lewismc** [comment]
@noah-de the following function within *drive.py* will be affected
```
    def mine_drive_urls_from_granule_search(self, granule_search_response=''):
        ''' Convenience function which extracts the PO.DAAC Drive URLs from \
           a given granule search obtained using podaac.granule_search(). \
           The response of this function is an array of strings denoting the \
           PO.DAAC Drive URLs to the granules.

           :param granule_search_response: the output response of a podaac.granule_search()
            :type path: :mod:`string`

            :returns: prints an array of PO.DAAC Drive URLs.
        '''
        from bs4 import BeautifulSoup
        soup = BeautifulSoup(granule_search_response, 'html.parser')
        drive_list = []
        for ftp_link in soup.find_all('link'):
            href = ftp_link.get('href')
            if 'ftp://podaac-ftp.jpl.nasa.gov/' in href:
                drive_list.append(href.replace('ftp://podaac-ftp.jpl.nasa.gov', self.URL))
            elif self.URL in href:
                drive_list.append(href)
        return drive_list
```
This is an easy fix however. I submitted it as part of the pull request in #140

**Comment by lewismc** [comment]
We actually need to implement unit tests for drive.py but I'll implement this in another ticket.

**Comment by lewismc** [comment]
Resolved in #140
